### ISSUE TITLE
importHelpers: true option doesn't import tslib and is still inlining the helpers

### ISSUE THREAD

**Issue by dmnsgn** [ezolenko/rollup-plugin-typescript2#58]
#### What happens and why it is wrong

I tried to compile a simple class with an async method:

```ts
class TestClass {
	static async fetchJson(url) {
		const response = await fetch(url);
		const data = await response.json();

		return { ...data }
	}
}
export { TestClass };
```
and I expect `"importHelpers": true,` to be respected and `tslib` to be added to the output. Instead I can see in the output that `__assign`, `__awaiter` and `__generator` are present.

Running `npx tsc src/index.ts --importHelpers --lib 'dom,ES2015' --target 'es5' --moduleResolution 'node' --declaration --noEmitHelpers` directly does work and generates `var tslib_1 = require("tslib");`.

#### Versions
- typescript:
```bash
$ npm ls typescript
test-class@2.2.0 /Users/damienseguin/Projects/#test/test-class
└── typescript@2.7.2
```
- rollup:
```bash
$ npm ls rollup
test-class@2.2.0 /Users/damienseguin/Projects/#test/test-class
└── rollup@0.56.3
```
- rollup-plugin-typescript2:
```bash
$ npm ls rollup-plugin-typescript2
test-class@2.2.0 /Users/damienseguin/Projects/#test/test-class
└── rollup-plugin-typescript2@0.11.1
```

#### rollup.config.js

```js
import resolve from "rollup-plugin-node-resolve";
import typescript from "rollup-plugin-typescript2";

export default {
  input: "src/index.ts",
  plugins: [
    resolve(),
    typescript({
      typescript: require("typescript"),
      clean: true,
      verbosity: 3
    }),
  ],
  output: [
    {
      format: "umd",
      file: "lib/test-class.js",
      name: "TestClass"
    },
    {
      format: "es",
      file: "lib/test-class.module.js"
    }
  ]
};
```

#### tsconfig.json

```json
{
  "compilerOptions": {
    "declaration": true,
    "importHelpers": true,
    "moduleResolution": "node",
    "target": "es5",
    "lib": ["dom", "es2015"]
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules",
    "**/*.spec.ts"
  ]
}
```

#### package.json

```json
{
  "name": "test-class",
  "main": "lib/test-class.js",
  "jsnext:main": "lib/test-class.module.js",
  "module": "lib/test-class.module.js",
  "types": "lib/index.d.ts",
  "scripts": {
    "dev": "rollup -c -w -m inline",
    "clean": "rm -rf lib",
    "build": "rollup -c"
  },
  "devDependencies": {
    "rollup": "^0.56.3",
    "rollup-plugin-node-resolve": "^3.0.3",
    "rollup-plugin-typescript2": "^0.11.1",
    "rollup-watch": "^4.3.1",
    "tslint": "^5.9.1",
    "typescript": "^2.7.2"
  },
  "dependencies": {
    "tslib": "^1.9.0"
  }
}
```

#### plugin output with verbosity 3
<details>
<summary>Output</summary>

```bash
src/index.ts → lib/test-class.js, lib/test-class.module.js...
rpt2: typescript version: 2.7.2
rpt2: rollup-plugin-typescript2 version: 0.11.1
rpt2: plugin options:
{
    "typescript": "version 2.7.2",
    "clean": true,
    "verbosity": 3,
    "check": true,
    "cacheRoot": "/Users/damienseguin/Projects/#test/test-class/.rpt2_cache",
    "include": [
        "*.ts+(|x)",
        "**/*.ts+(|x)"
    ],
    "exclude": [
        "*.d.ts",
        "**/*.d.ts"
    ],
    "abortOnError": true,
    "rollupCommonJSResolveHack": false,
    "useTsconfigDeclarationDir": false,
    "tsconfigOverride": {},
    "tsconfigDefaults": {}
}
rpt2: rollup config:
{
    "input": "src/index.ts",
    "plugins": [
        {
            "name": "node-resolve"
        },
        {
            "name": "rpt2"
        }
    ],
    "entry": "src/index.ts",
    "external": [
        "",
        ""
    ]
}
rpt2: built-in options overrides: {
    "module": 5,
    "noEmitHelpers": false,
    "importHelpers": true,
    "noResolve": false,
    "noEmit": false,
    "outDir": "/Users/damienseguin/Projects/#test/test-class",
    "moduleResolution": 2,
    "declarationDir": "/Users/damienseguin/Projects/#test/test-class"
}
rpt2: parsed tsconfig: {
    "options": {
        "declaration": true,
        "importHelpers": true,
        "moduleResolution": 2,
        "target": 1,
        "lib": [
            "lib.dom.d.ts",
            "lib.es2015.d.ts"
        ],
        "module": 5,
        "noEmitHelpers": false,
        "noResolve": false,
        "noEmit": false,
        "outDir": "/Users/damienseguin/Projects/#test/test-class",
        "declarationDir": "/Users/damienseguin/Projects/#test/test-class",
        "configFilePath": "/Users/damienseguin/Projects/#test/test-class/tsconfig.json"
    },
    "fileNames": [
        "/Users/damienseguin/Projects/#test/test-class/src/index.ts",
        "/Users/damienseguin/Projects/#test/test-class/src/types.ts"
    ],
    "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
    },
    "raw": {
        "compilerOptions": {
            "declaration": true,
            "importHelpers": true,
            "moduleResolution": "node",
            "target": "es5",
            "lib": [
                "dom",
                "es2015"
            ]
        },
        "include": [
            "src/**/*"
        ],
        "exclude": [
            "node_modules",
            "**/*.spec.ts"
        ]
    },
    "errors": [],
    "wildcardDirectories": {
        "/users/damienseguin/projects/#test/test-class/src": 1
    },
    "compileOnSave": false,
    "configFileSpecs": {
        "includeSpecs": [
            "src/**/*"
        ],
        "excludeSpecs": [
            "node_modules",
            "**/*.spec.ts"
        ],
        "validatedIncludeSpecs": [
            "src/**/*"
        ],
        "validatedExcludeSpecs": [
            "node_modules",
            "**/*.spec.ts"
        ],
        "wildcardDirectories": {
            "/users/damienseguin/projects/#test/test-class/src": 1
        }
    }
}
rpt2: included:
'[
    "*.ts+(|x)",
    "**/*.ts+(|x)"
]'
rpt2: excluded:
'[
    "*.d.ts",
    "**/*.d.ts"
]'
rpt2: Ambient types:
rpt2: cleaning cache: /Users/damienseguin/Projects/#test/test-class/.rpt2_cache/7247f35da223bc044adbbddbc34442c3554b0160
rpt2: transpiling '/Users/damienseguin/Projects/#test/test-class/src/index.ts'
rpt2:     cache: '/Users/damienseguin/Projects/#test/test-class/.rpt2_cache/7247f35da223bc044adbbddbc34442c3554b0160/code/cache/efbe672c2695ac546f259f206ba52ed5aca5d46b'
rpt2:     cache miss
rpt2:     cache: '/Users/damienseguin/Projects/#test/test-class/.rpt2_cache/7247f35da223bc044adbbddbc34442c3554b0160/syntacticDiagnostics/cache/efbe672c2695ac546f259f206ba52ed5aca5d46b'
rpt2:     cache miss
rpt2:     cache: '/Users/damienseguin/Projects/#test/test-class/.rpt2_cache/7247f35da223bc044adbbddbc34442c3554b0160/semanticDiagnostics/cache/efbe672c2695ac546f259f206ba52ed5aca5d46b'
rpt2:     cache miss
rpt2: generated declarations for '/Users/damienseguin/Projects/#test/test-class/src/index.ts'
rpt2: generating target 1
rpt2: rolling caches
rpt2: generating missed declarations for '/Users/damienseguin/Projects/#test/test-class/src/types.ts'
rpt2: writing declarations for '/Users/damienseguin/Projects/#test/test-class/src/index.ts' to '/Users/damienseguin/Projects/#test/test-class/lib/index.d.ts'
rpt2: writing declarations for '/Users/damienseguin/Projects/#test/test-class/src/types.ts' to '/Users/damienseguin/Projects/#test/test-class/lib/types.d.ts'
rpt2: generating target 2
rpt2: rolling caches
rpt2: writing declarations for '/Users/damienseguin/Projects/#test/test-class/src/index.ts' to '/Users/damienseguin/Projects/#test/test-class/lib/index.d.ts'
rpt2: writing declarations for '/Users/damienseguin/Projects/#test/test-class/src/types.ts' to '/Users/damienseguin/Projects/#test/test-class/lib/types.d.ts'
created lib/test-class.js, lib/test-class.module.js in 629ms
```

</details>

**Comment by ezolenko** [comment]
This is by design: the idea is that if you are using rollup you expect all or most dependencies to be rolled into a single bundle. 

Do you have a usecase for keeping tslib out of your bundle?

**Comment by dmnsgn** [comment]
If I get it right the whole purpose of `tslib` is to avoid duplicating the typescript helpers in every file that needs them.

Rollup allows me to output two bundles to be consumed as:
* an es module eg. `import { TestClass } from "test-class"`
* an `UMD` file:
  * commonjs eg. `const { TestClass } = require("test-class")`
  * AMD require js (something like `define(['TestClass'])` I haven't used in years so I don't remember the syntax)
  * with the browser's `window` directly eg. ```<script src="vendors/test-class.js"></script><script>const testClass = new window.TestClass();</script>```

I expect my bundle to contain the imports to any dependency because:
* the es module bundle will be targeted at environment that read the es module syntax (so `import * as tslib_1 from "tslib";` will be understood)
* the UMD bundle will be _Universal_ so node, requirejs and the browser could read it with the umd factory function (so `var tslib_1 = require("tslib");` or `define['tslib]` or `window.tslib` will be understood)

In other words, if my dependencies are specified in my `package.json` `dependencies` key, I don't want them to be bundled and included by rollup in my resulting build but I want their import syntax to be different depending on the environment I am targeting.

**Comment by dmnsgn** [comment]
Actually yes, it seems that setting `external: ["tslib"]` in my `rollup.config.js` is doing what I am describing in the previous comment.

**Comment by ezolenko** [comment]
Ah yes, `external` should work.
